swarm/src/behaviour/external_addresses.rs

```diff
@@ -0,0 +1,43 @@
+use crate::behaviour::{ExpiredExternalAddr, FromSwarm, NewExternalAddr};
+use crate::IntoConnectionHandler;
+use libp2p_core::Multiaddr;
+use std::collections::HashSet;
+
+/// Utility struct for tracking the external addresses of a [`Swarm`](crate::Swarm).
+#[derive(Debug, Default, Clone)]
+pub struct ExternalAddresses {
```

Comment:
```suggestion
pub struct ExternalAddressTracker {
```

Not a strong opinion, though this would imply that one has to actively interact / pass-events with/to it.

Reply:
I am not convinced that this is a better name. Can you elaborate what you don't like about the current one? :)

Reply:
The name `ExternalAddresses` does not imply to me as a user, that I need to pass the `FromSwarm` events into it. That said, I think `ExternalAddresses` is just fine.

Reply:
I would argue that this is only something a user needs to learn once and quite obvious from the implementation.

Most of the time, this name should convey that it holds the external addresses and only every so often, you need to understand how it works so I optimised for the common bit :)